### PR TITLE
Inliner: enable xml format dump for inlines

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Add new config option to dump inline trees as XML, and implement
dumping support.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@kyulee PTAL. Preliminary step towards doing some automated inline space exploration.

Also reworked some of the other dumping code to use %* formatting.

cc @dotnet/jit-contrib

File: src/jit/inline.cpp [hunk 1]
```diff
@@ -1120,83 +1168,159 @@ bool InlineStrategy::s_DumpDataHeader = false;
...
+#else
+    unsigned hash = 0;
+#endif
+
+    // Convert time spent jitting into microseconds
+    unsigned microsecondsSpentJitting = 0;
+    unsigned __int64 compCycles = m_Compiler->getInlineCycleCount();
+    if (compCycles > 0)
+    {
+        double countsPerSec = CycleTimer::CyclesPerSecond();
```
Comment (kyulee1) [inline src/jit/inline.cpp]: It appears this could be 0 although it's unlikely. Should we bail out computing time in such case?

**Comment by kyulee1** [comment]
LGTM with a minor comment.

**Comment by AndyAyersMS** [comment]
Some new infrastructure issue is doing in most of the CI runs. @mmitche ever seen this?
```
00:50:36.541 Archiving artifacts
00:51:40.449 FATAL: 0
00:51:40.449 java.lang.ArrayIndexOutOfBoundsException: 0
00:51:40.449 	at hudson.plugins.git.GitChangeSet.findOrCreateUser(GitChangeSet.java:377)
```

File: src/jit/inline.cpp (hunk 1 above)
Comment (AndyAyersMS) [inline src/jit/inline.cpp]: Various checks over on the compiler side ensure `compCycles` is zero if the cycles per second can't be determined. See for instance `Compiler::RecordStateAtEndOfCompilation`

**Comment by AndyAyersMS** [comment]
Am going to hope it's a glitch and just retest.

@dotnet-bot retest this please

**Pull request merged**